internal/parser/planparserv2/utils.go

```diff
@@ -787,3 +788,77 @@ func decodeUnicode(input string) string {
 		return string(rune(code))
 	})
 }
+
+func parseISODuration(durationStr string) (*planpb.Interval, error) {
+	iso8601DurationRegex := regexp.MustCompile(
+		`^P` + // P at the start
+			`(?:(\d+)Y)?` + // Years (optional)
+			`(?:(\d+)M)?` + // Months (optional)
+			`(?:(\d+)D)?` + // Days (optional)
+			`(?:T` + // T separator (optional, but required for time parts)
+			`(?:(\d+)H)?` + // Hours (optional)
+			`(?:(\d+)M)?` + // Minutes (optional)
+			`(?:(\d+)S)?` + // Seconds (optional)
+			`)?$`,
+	)
+	matches := iso8601DurationRegex.FindStringSubmatch(durationStr)
+	if matches == nil {
+		return nil, fmt.Errorf("invalid ISO 8601 duration: %s", durationStr)
+	}
+
+	interval := &planpb.Interval{}
+	if matches[1] != "" {
+		years, _ := strconv.ParseInt(matches[1], 10, 64)
```

Comment:
The errors returned by strconv.ParseInt are currently ignored (_). If a numerical value in the input string exceeds the int64 maximum or contains non-digit characters, ParseInt will return an error. 

    years, err := strconv.ParseInt(matches[1], 10, 64)
    if err != nil {
        return nil, fmt.Errorf("invalid year value in duration: %w", err)
    }
    interval.Years = years